### PR TITLE
[Feature] github show user org repos [OSF-1466]

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Allow users to link their github organizations repos to osf project.

## Changes

- Add github user organizations repos to select repo modal.
- Update github addon to use github3 1.0.0a3
- Update tests

## Ticket

[OSF-1466](https://openscience.atlassian.net/browse/OSF-1466)

## After
3rz0cuf7j3

**screenshot**
screen shot 2018-02-22 at 2 40 34 pm

### CODE REVIEW

File: addons/github/models.py [hunk 1]
```diff
@@ -197,22 +197,19 @@ def to_json(self, user):
...
                     '{0} / {1}'.format(repo.owner.login, repo.name)
                     for repo in repos
-                ]
+                }
             except GitHubError:
                 repo_names = []
                 valid_credentials = False
             if owner == user:
-                ret.update({'repo_names': repo_names})
+                ret.update({'repo_names': list(repo_names)})
```
Comment (mfraezz) [inline addons/github/models.py]: This `set`/`list` conversion effectively dedupes it, but scrambles the ordering. You can use `sorted` to fix that.

File: addons/github/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 cachecontrol==0.10.2
-github3.py==0.9.0
+github3.py==1.0.0a3
```
Comment (mfraezz) [inline addons/github/requirements.txt]: Latest version is `1.0.0a4`. Any reason not to pin to that one? See [changelog](http://github3py.readthedocs.io/en/latest/project_changelog.html)

Comment (fabmiz) [reply addons/github/requirements.txt]: I will go ahead and bump it to `1.0.0a4`. Initially, the goal was to bump up enough versions to solve the issue.

File: website/templates/project/addons.mako [hunk 3]
```diff
@@ -116,6 +116,26 @@
                         % if addon.get('node_settings_template'):
                             ${render_node_settings(addon)}
                         % endif
+                        % if addon['addon_short_name'] == 'github':
+                            <div class="dismissible-alerts hidden" data-bind="css: {'hidden': $root.isDismissed('githubOrgs')}">
+                                <div class="alert alert-info alert-dismissible" role="alert">
+                                    <button type="button" class="close" data-dismiss="alert" aria-label="Close"
+                                        data-bind="tooltip: {title: 'Once dismissed, this alert will not be shown again.', placement: 'left'}, click: $root.dismiss.bind($root, 'githubOrgs')">
```
Comment (sloria) [inline website/templates/project/addons.mako]: This tooltip is unnecessary--as a user, I would expect clicking the "x" to dismiss the alert, so this just adds noise.

File: website/templates/project/addons.mako [hunk 4]
```diff
@@ -116,6 +116,26 @@
...
+                                        <span aria-hidden="true">&times;</span>
+                                    </button>
+                                    <div>
+                                        <h4>Don’t see your Github Organization repositories?</h4>
```
Comment (sloria) [inline website/templates/project/addons.mako]: "GitHub", not "Github".

File: website/templates/project/addons.mako [hunk 5]
```diff
@@ -116,6 +116,26 @@
...
+                                        <p>
+                                            You may need to reauthorize your Github access token.
+                                            Follow the steps in the <a class="alert-link" href="http://help.osf.io/a/850865-reauthorize-github" target="_black">help guide</a> to resolve the issue. <br>
+                                        </p>
+                                        <p>
+                                            Please contact support <a class="alert-link" href="mailto:contact@osf.io">support@osf.io</a> if you have questions.
```
Comment (pattisdr) [inline website/templates/project/addons.mako]: I assume the mailto: should be support@osf.io.  Also, is the word support needed?
Mine looks like this:
`Please contact <a class="alert-link" href="mailto:support@osf.io">support@osf.io</a> if you have questions.`

**Comment by chennan47** [comment]
@fabmiz some merge conflicts need to be solved.

File: docker-sync.yml [hunk 6]
```diff
@@ -24,13 +24,13 @@ syncs:
 #    sync_excludes: ['.DS_Store', '*.pyc', '*.tmp', '.git', '.idea']
 #    watch_excludes: ['.*\.DS_Store', '.*\.pyc', '.*\.tmp', '.*/\.git', '.*/\.idea']
 
-#  emberosf-sync:
```
Comment (chennan47) [inline docker-sync.yml]: Those docker change doesn't seem to be related to the purpose of this pr. Are those necessary?

File: docker-sync.yml [hunk 7]
```diff
@@ -40,13 +40,13 @@ syncs:
...
 #    watch_excludes: ['.*\.DS_Store', '.*\.map', '.*\.pyc', '.*\.tmp', '.*/\.git', '.*/\.idea', '.*/bower_components', '.*/node_modules', '.*/tmp', '.*/dist']
 
-#  preprints-sync:
-#    src: '../ember-osf-preprints'
-#    sync_strategy: 'native_osx'
-#    sync_args: [ '-prefer newer' ]
-#    sync_excludes_type: 'Name'
-#    sync_excludes: ['.DS_Store', '*.map', '*.pyc', '*.tmp', '.git', '.idea', 'bower_components', 'node_modules', 'tmp', 'dist']
-#    watch_excludes: ['.*\.DS_Store', '.*\.map', '.*\.pyc', '.*\.tmp', '.*/\.git', '.*/\.idea', '.*/bower_components', '.*/node_modules', '.*/tmp', '.*/dist']
+  preprints-sync:
```
Comment (chennan47) [inline docker-sync.yml]: Same as above.

File: docker-compose.override.yml [hunk 8]
```diff
@@ -73,27 +73,27 @@ services:
...
-##      - cd /ember-osf &&
-##        yarn link &&
-##        cd /code &&
-##        (rm -r node_modules || true) &&
-##        yarn --frozen-lockfile &&
-##        yarn link @centerforopenscience/ember-osf &&
-##        (rm -r bower_components || true) &&
-##        ./node_modules/.bin/bower install --allow-root --config.interactive=false &&
-##        yarn start --host 0.0.0.0 --port 4201 --live-reload-port 41954
+  preprints:
```
Comment (chennan47) [inline docker-compose.override.yml]: Those docker change doesn't seem to be related to the purpose of this pr. Are those necessary?

Comment (fabmiz) [reply docker-compose.override.yml]: No, I accidentally committed these.

**Comment by fabmiz** [comment]
@chennan47, courtesy of my editor, this latest PR removes unnecessary whitespaces in docker-*.yml. Though unrelated, they are harmless and help enforce proper formatting rules.

**Comment by mfraezz** [comment]
Merged via 078c5d38e6c61f972cb38968154bc46eba91274c, closing
